src/servicex_did_finder_lib/logging.py

```diff
@@ -0,0 +1,45 @@
+import logging
+import os
+
+
+class DIDFormatter(logging.Formatter):
+    """
+    Need a customer formatter to allow for logging with request ids that vary.
+    Normally log messages are "level instance component request_id msg" and
+    request_id gets set by initialize_logging but we need a handler that'll let
+    us pass in the request id and have that embedded in the log message
+    """
+
+    def format(self, record: logging.LogRecord) -> str:
+        """
+        Format record with request id if present, otherwise assume None
+
+        :param record: LogRecord
+        :return: formatted log message
+        """
+
+        if hasattr(record, "requestId"):
+            return super().format(record)
+        else:
+            setattr(record, "requestId", None)
+            return super().format(record)
+
+
+def initialize_root_logger(did_scheme: str):
+    """
+    Get a logger and initialize it so that it outputs the correct format
+    :param request: Request id to insert into log messages
```

Comment:
Can you update the docstring to reflect the new parameter?

Reply:
Thanks for catching that!